README.adoc

```diff
@@ -0,0 +1,223 @@
+====
+https://github.com/rubocop-hq/rubocop[RuboCop], a static code analyzer (linter) and formatter,
+has a https://github.com/utkarsh2102/rubocop-packaging[`rubocop-packaging`] extension, based
+on this style guide.
+
+[source,shell]
+----
+gem install rubocop-packaging
+----
+====
+
+
+== How To Read This Guide
+
+The guide is separated into sections based on the different cops that the Packaging extension
+provides. +
+There was an attempt to explicitly mention everything so if anything is still unclear, feel
+free to open an issue asking for further clarity.
+
+
+== A Living Document
+
+This guide is a work in progress - existing guidelines are constantly being improved, new
+guidelines are being added, and occasionally some guidelines would get removed.
+
+
+== Why Packaging Extension?
+
+The Debian Ruby team has a lot of experience in packaging and maintaining Ruby libraries and
+applications for Debian. During this work, they identified several issues in upstream codebases
+that makes it difficult to build a Debian package straight out of those Ruby gems (shipped via
+https://rubygems.org[rubygems]).
+
+The Debian developers (downstream maintainers) have been in touch with the RubyGems and other
+upstream maintainers and we're collaborating to try to make things easier for OS packagers
+while not compromising the experience for upstream maintainers.
+
+As a result, we're working on this RuboCop extension to enforce a set of best practices that
+upstream maintainers can follow to make lifes of packagers easier. +
+And that is how `rubocop-packaging` is born!
+
+
+== Cops Description
+
+Listed below are the cops provided by the `Packaging` extension, describing the need to write
+and enforce it and how is it problematic on the Debian (downstream) side.
+
+=== Gemspec Git [[gemspec-git]]
+
+About the `GemspecGit` cop:
+
+==== Rationale [[gemspec-git-rationale]]
+
+Avoid using `git ls-files` to produce lists of files. Downstreams often need to build your
+package in an environment that does not have git (on purpose). Instead, use some
+pure Ruby alternative, like `Dir` or `Dir.glob`.
+
+==== Need [[gemspec-git-need]]
+
+Packages in Debian are built in a clean environment (https://wiki.debian.org/sbuild[sbuild],
+https://wiki.debian.org/Schroot[schroot], et al) and whilst doing so, the build fails with: +
+`Invalid gemspec in [<gem_name>.gemspec]: No such file or directory - git`
+
+And adding `git` as a dependency for each of the Ruby packaging is something that is not right
+and definitely not recommended.
+
+Therefore, the other way is to patch out the usage of `git` and use some plain Ruby alternative
+like `Dir` or `Dir.glob` or even `Rake::FileList`.
+
+There's not only Debian or other OS packaging situation/examples, but also a couple others,
+for instance:
+
+* `ruby-core` as part of their CI system runs their test suite against an unpackaged Ruby
+  tarball which doesn't have a `.git` folders. That means they needed to overwrite the
```

Comment:
a `.git` folder (singular) - or may be use directory instead of folder as folder is usually referred in graphical environments I think.

Reply:
Great catch, fixed!